src/components/ChatRoomList.js

```diff
@@ -0,0 +1,170 @@
+import React, { useState, useEffect } from 'react';
+import { PlusCircleFill } from 'react-bootstrap-icons';
+import { Link, useHistory } from 'react-router-dom';
+import styled from 'styled-components';
+import { useContacts } from '../contexts/ContactsProvider';
+import { useRooms } from '../contexts/RoomsProvider';
+import ChatRoomListItem from './ChatRoomListItem';
+
+const StyledChatRoomList = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  margin: 0;
+`;
+
+const Actions = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  margin-bottom: 20px;
+  padding: 0 15px;
+  height: 40px;
+`;
+
+const SearchInput = styled.input`
+  border: none;
+  outline: none;
+  padding: 0 15px;
+  height: 100%;
+  border-radius: 5px;
+  background: #e3e3e3;
+  font-size: 1em;
+  text-align: center;
+  flex: 1 1 auto;
+  &:focus {
+    text-align: left;
+  }
+`;
+
+const CreateChatButton = styled.button`
+  margin: 0 0 0 15px;
+  background: #c6d8fc;
+  outline: none;
+  border: none;
+  border-radius: 5px;
+  padding: 0 15px;
+  height: 100%;
+  box-shadow: none;
+  font-size: 1.3rem;
+  display: flex;
+  color: #0e388a;
+  cursor: pointer;
+  justify-content: center;
+  align-items: center;
+  & span {
+    margin-left: 0.8rem;
+    display: none;
+  }
+  &:hover {
+    font-size: 1rem;
+  }
+  &:hover span {
+    display: block;
+  }
+`;
+
+const ChatRoomListItems = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex: 1 1 auto;
+  padding: 0 15px;
+  width: 100%;
+`;
+
+export default function ChatRoomList() {
+  const history = useHistory();
+  const { currentUser } = useContacts();
+  const { currentUserRooms, createRoom } = useRooms();
+  const [searchQuery, setSearchQuery] = useState('');
+  const [filteredRooms, setFilteredRooms] = useState(currentUserRooms);
+
+  useEffect(() => {
+    setFilteredRooms(() =>
+      currentUserRooms.filter((room) =>
+        room.roomName.toLowerCase().includes(searchQuery.toLowerCase())
+      )
+    );
+  }, [currentUserRooms, searchQuery]);
+
+  const handleSearchQueryChange = (e) => {
```

Comment:
```suggestion
  const handleSearchInputChange = (e) => {
```

이벤트가 정확이 어떤 것인지 변수명에서 확실히 정하는 습관을 가집시다 ㅎㅎ